paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -152,27 +129,17 @@ public class PaparazziPlugin : Plugin<Project> {
         val gradleHomeDir = projectDirectory.dir(project.gradle.gradleUserHomeDir.path)
 
         task.packageName.set(android.packageName())
-        task.artifactFiles.from(packageAwareArtifactFiles)
+        task.artifactFiles.from(sources.packageAwareArtifactFiles)
         task.nonTransitiveRClassEnabled.set(nonTransitiveRClassEnabled)
         task.targetSdkVersion.set(android.targetSdkVersion())
-
-        val localResourcePaths = localResourceDirs
-          .map { layers -> layers.flatten() }
-          .map { dirs -> dirs.map { projectDirectory.relativize(it.asFile) }.asReversed() }
-
-        task.projectResourceDirs.set(localResourcePaths)
-        task.moduleResourceDirs.set(moduleResourceDirs.relativize(projectDirectory))
-        task.aarExplodedDirs.set(aarExplodedDirs.relativize(gradleHomeDir))
+        task.projectResourceDirs.set(sources.localResourceDirs.relativize(projectDirectory))
+        task.moduleResourceDirs.set(sources.moduleResourceDirs.relativize(projectDirectory))
+        task.aarExplodedDirs.set(sources.aarExplodedDirs.relativize(gradleHomeDir))
         task.projectAssetDirs.set(
-          run {
-            val localAssetPaths = localAssetDirs
-              .map { layers -> layers.flatten() }
-              .map { dirs -> dirs.map { projectDirectory.relativize(it.asFile) }.asReversed() }
-            val moduleAssetPaths = moduleAssetDirs.relativize(projectDirectory)
-            localAssetPaths.zip(moduleAssetPaths, List<String>::plus)
-          }
+          sources.localAssetDirs.relativize(projectDirectory)
+            .zip(sources.moduleAssetDirs.relativize(projectDirectory), List<String>::plus)
```

Comment:
minor, but this feels like:

minor, but this might be cleaner (if compiles)

```suggestion
          sources.localAssetDirs.zip(sources.moduleAssetDirs, List<String>::plus)
            .relativize(projectDirectory)
```

Reply:
yea, i thought of that, but didn't want to push the providerFactory into AndroidVariantSource in order to handle the nullability of the localAssetDirs and zip doesn't take a nullable receiver.  Might just be worth splitting this out into two separate task properties as we do for resources, but I think this is fine to land for now.